torch/sparse/__init__.py

```diff
@@ -201,3 +201,154 @@ def log_softmax(input: Tensor, dim: int, dtype: Optional[DType] = None) -> Tenso
+to the boolean tensor :attr:`mask`.''',
+        masked_reduction_args='''\
+If :attr:`keepdim` is ``True``, the output tensor is of the same
+size as :attr:`input` except in the dimension(s) :attr:`dim` where
+it is of size 1.  Otherwise, :attr:`dim` is squeezed (see
+:func:`torch.squeeze`), resulting in the output tensor having 1
+fewer dimension.
+
+The boolean tensor :attr:`mask` defines the "validity" of
+:attr:`input` tensor elements: if :attr:`mask` element is True
+then the corresponding element in :attr:`input` tensor will be
+included in {function name} computation, otherwise the element is
+ignored.
+
+When all elements of :attr:`input` along the given dimension
+:attr:`dim` are ignored, the corresponding element of the output
+tensor will have undefined value: it may or may not correspond to the
+identity value of {operation name} operation; the choice may
+correspond to the value that leads to the most efficient storage of
+:attr:`output` tensor.
+
+The shapes of the :attr:`mask` tensor and the :attr:`input` tensor
+don't need to match, but they must be :ref:`broadcastable
+<broadcasting-semantics>`.
+
+Args:
+    input (Tensor): the input tensor
+    dim (int): a dimension along which {function name} will be
+      computed.
+
+Keyword args:
+    keepdim (bool, optional): whether the output tensor has
+      :attr:`dim` retained or not. Default: False.
+    dtype (:class:`torch.dtype`, optional): the desired data type
+      of returned tensor.  If specified, the input tensor is
+      casted to :attr:`dtype` before the operation is
+      performed. Default: None
+    mask (:class:`torch.Tensor`, optional): the boolean tensor
+      containing the binary mask of validity of input tensor
+      elements.
+      Default: ``torch.ones(input.shape, dtype=torch.bool)``.''',
+        masked_reduction_example='''\
+Example::
+
+    >>> input = torch.tensor([[-3, -2, -1], [0, 1, 2]])
+    >>> input
+    tensor([[-3, -2, -1],
+            [ 0,  1,  2]])
+    >>> mask = torch.tensor([[True, False, True], [False, False, False]])
+    >>> mask
+    tensor([[ True, False,  True],
+            [False, False, False]])''',
+        masked_normalization_args='''\
+    TBD
+''')
+
+    # Apply function name info to docstring templates:
+    templates = dict(
+        (k, v.format_map(
+            {'function_name': func.__name__,
+             'function name': func.__name__.replace('_', ' '),
+             'operation_name': '_'.join(func.__name__.split('_')[1:]),
+             'operation name': ' '.join(func.__name__.split('_')[1:])}
+        )) for k, v in docstring_templates.items())
+
+    # Apply docstring templates to function doctring:
+    func.__doc__ = func.__doc__.format_map(templates)
+
+    # Expose function as public symbol
+    __all__.append(func.__name__)
+
+    return func
+
+
+@_apply_docstring_templates
+def masked_sum(input: Tensor,
+               mask: Optional[Tensor] = None,
```

Comment:
Why not add it to the end? That way it'll be a replacement positionally as well (I think)

Reply:
The answer depends on how tied are we to the definition of reduction operations by OpInfo:
https://github.com/pytorch/pytorch/blob/5060b69d62943e11f053f8c86b8154a7d009af11/torch/testing/_internal/common_methods_invocations.py#L860-L865
and how to interpret the usage of `*args` and `**kwargs` here.
I don't have strong preferences but it seemed to me that putting `mask` to `*args` part would be a first natural choice.
Following https://numpy.org/doc/stable/reference/generated/numpy.sum.html?highlight=sum#numpy.sum, the end position would be more appropriate indeed.

Reply:
To me it's a keyword argument in the sense that it's optional. By default it is ignored. I suggest we get another person involved here to see how to best make a decision after we've revised the other parts of this PR.